server/chat-commands/core.ts

```diff
@@ -732,12 +788,7 @@ export const commands: ChatCommands = {
 			}
 			for (const setting in user.settings) {
 				if (setting in raw) {
-					if (setting === 'blockPMs' &&
-						Users.Auth.isAuthLevel(raw[setting])) {
-						settings[setting] = raw[setting];
-					} else {
-						settings[setting as keyof UserSettings] = !!raw[setting];
-					}
+					settings[setting as keyof UserSettings] = typeof raw[setting] === 'object' ? raw[setting] : !!raw[setting];
```

Comment:
you still need to check if `Users.Auth.isAuthLevel`... the whole point of this code is to not blindly expect potentially malicious input...